test/e2e/storage/volume_expand_online.go

```diff
@@ -0,0 +1,312 @@
+		}
+
+		By("Waiting for cloudprovider new resize to finish")
+		err = waitForControllerVolumeResize(pvc, c)
+		Expect(err).NotTo(HaveOccurred(), "While waiting for pvc resize to finish")
+
+		By("Checking for latest conditions on pvc")
+		pvc, err = c.CoreV1().PersistentVolumeClaims(ns).Get(pvc.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "While fetching pvc after controller resize")
+
+		latestInProgressConditions := pvc.Status.Conditions
+		Expect(len(latestInProgressConditions)).To(Equal(1), "pvc must have file system resize pending condition")
+		Expect(latestInProgressConditions[0].Type).To(Equal(v1.PersistentVolumeClaimFileSystemResizePending), "pvc must have fs resizing condition")
+
+		By("Waiting for file system resize to finish")
+		pvc, err = waitForFSResize(pvc, c)
+		Expect(err).NotTo(HaveOccurred(), "while waiting for fs resize to finish")
+
+		pvcConditions := pvc.Status.Conditions
+		Expect(len(pvcConditions)).To(Equal(0), "pvc should not have conditions")
+	})
+
+	It("Try resizing PVC when AllowVolumeExpansion is not set in sc", func() {
+		By("Creating storageclass without AllowVolumeExpansion parameter")
+		c = f.ClientSet
+		ns = f.Namespace.Name
+		test := storageClassTest{
+			name:      "default",
+			claimSize: "2Gi",
+		}
+		nonresizableSc, err = createNonResizableStorageClass(test, ns, "nonresizing", c)
+		Expect(err).NotTo(HaveOccurred(), "Error creating non-resizable storage class")
+
+		pvc = newClaim(test, ns, "default")
+		pvc.Spec.StorageClassName = &nonresizableSc.Name
+		pvc, err = c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(pvc)
+		Expect(err).NotTo(HaveOccurred(), "Error creating pvc")
+
+		defer func() {
+			framework.ExpectNoError(framework.DeletePersistentVolumeClaim(c, pvc.Name, pvc.Namespace))
+			framework.ExpectNoError(c.StorageV1().StorageClasses().Delete(nonresizableSc.Name, nil))
+		}()
+
+		By("Waiting for pvc to be in bound phase")
+		pvcClaims := []*v1.PersistentVolumeClaim{pvc}
+		pvs, err := framework.WaitForPVClaimBoundPhase(c, pvcClaims, framework.ClaimProvisionTimeout)
+		Expect(err).NotTo(HaveOccurred(), "Failed waiting for PVC to be bound %v", err)
+		Expect(len(pvs)).To(Equal(1))
+
+		By("Trying to expand current pvc")
+		newSize := resource.MustParse("4Gi")
+		pvc, err = expandPVCSize(pvc, newSize, c)
+		framework.Logf("got err %v", err)
+		Expect(err).To(HaveOccurred())
+
+		pvcSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+		if pvcSize.Cmp(newSize) == 0 {
+			framework.Failf("AllowVolumeExpansion is False, resizing pvc is not allowed %q", pvc.Name)
+		}
+
+		pvcConditions := pvc.Status.Conditions
+		Expect(len(pvcConditions)).To(Equal(0), "pvc should not have conditions")
+	})
+
+	It("Try resizing PVC that is in pending state", func() {
+		scName := getDefaultStorageClassName(c)
+		test := storageClassTest{
+			name:      "default",
+			claimSize: "2Gi",
+		}
+
+		By("setting the is-default StorageClass annotation to false")
```

Comment:
I don't quite follow what you're doing with the storage classes here. Can't you just create a new dummy storage class instead of turning on/off defaults.

Reply:
tests can be run in parallel so you want to avoid messing with shared objects in tests